operator/controllers/actionsrunner/util/filters.go

```diff
@@ -45,3 +45,21 @@ func FilterEnv(oldEnv []corev1.EnvVar, predicate func(corev1.EnvVar) bool) []cor
 
 	return newEnv
 }
+
+func FilterResources(source corev1.ResourceRequirements, resourceNames ...corev1.ResourceName) corev1.ResourceRequirements {
+	resourceRequirements := corev1.ResourceRequirements{
+		Limits:   corev1.ResourceList{},
+		Requests: corev1.ResourceList{},
+	}
+
+	for _, resourceName := range resourceNames {
+		if val, ok := source.Limits[resourceName]; ok {
```

Comment:
Does source ever come null? You may want to add a guard clause at the beginning of method

Reply:
No, it doesn't. `source` is a struct, not a pointer, then it is always valid.